### DISCUSSION TITLE
How can I configure Vitest so that it doesn't create a `node_modules` directory in my `app/javascript` directory?

### DISCUSSION THREAD

**Discussion by jameshibbard** [vitest-dev/vitest#2019] [Q&A] [answered]
Hi,

I'm using Vitest in conjunction with Vue Test Utils to test Vue components in a Rails app.
Whenever I run Vitest, it creates a `node_modules` folder in the folder holding my JavaScript (`app/javascript`).
Is there any way to prevent this?

My (abridged) directory structure looks like this:

```
.
├── app
│    └── javascript
│        ├── components
│        └── entrypoints
│            └── application.js
├── node_modules
├── test
│     └── javascript
│         └── components
│             └── whatever.test.js
└── vite.config.ts
```

This is what things look like after running Vitest (via an npm script):

```
.
├── app
│   └── javascript
│       ├── components
│       ├── entrypoints
│       └── node_modules
│           └── .vitest
│               └── results.json
```

It seems as though Vitest considers `app/javascript` to be the root of all things. 

Is there any way to configure Vitest so that it creates this `.vitest` folder in the `node_modules` directory at the root of my app and not in `app/javascript`?

---

In case it helps:

I'm using Vitest v0.23.1 and [Vite Ruby](https://vite-ruby.netlify.app/) to take care of the bundling.

My vite.config.ts file looks like this:

```ts
export default defineConfig({
  plugins: [ ... ],
  test: {
    environment: 'jsdom',
    include: ['../../test/javascript/components/*.test.js'],
  },
});
```

**Comment by sheremet-va** [comment, marked as answer]
Use `test.cache` option (you can disable it or make path different)

**Reply by jameshibbard** [reply]
Thank you. That worked perfectly.

**Comment by devrizwanali** [comment]
@jameshibbard - Hi 

Are you using any Ruby code in the files that are used in tests?

I'm trying to setup vitest with my ROR+vue project. But in tests, I'm using some of the ruby code written in `.js.erb` files, the tests were working fine with Webpacker. Now I'm upgrading from webpacker to vite.

**Reply by jameshibbard** [reply]
> Are you using any Ruby code in the files that are used in tests?

Nope. Just JavaScript.
